components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -684,6 +686,32 @@ public API addAPI(API api) throws APIManagementException {
             handleException("Error occurred while adding default API LifeCycle.", e);
         }
 
+        TreeMap<String, API> apiSortedMap = new TreeMap<>();
+
+        List<API> apiList = getAPIVersionsByProviderAndName(provider, apiName, api.getOrganization());
+            for (API mappedAPI : apiList) {
+                apiSortedMap.put(mappedAPI.getVersionTimestamp(), mappedAPI);
+            }
+
+            APIVersionStringComparator comparator = new APIVersionStringComparator();
+            String latestVersion = api.getId().getVersion();
+            long previousTimestamp = 0L;
+            String latestTimestamp = "";
+            for (API tempAPI : apiSortedMap.values()) {
+                if (comparator.compare(tempAPI.getId().getVersion(), latestVersion) > 0){
+                    latestTimestamp = String.valueOf((previousTimestamp + Long.valueOf(tempAPI.getVersionTimestamp()))/2) ;
+                    break;
+                } else {
+                    previousTimestamp = Long.valueOf(tempAPI.getVersionTimestamp());
+
+                }
+
+            }
```

Comment:
remove extra spaces

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/10913